### PR TITLE
[BCI-3500] Modify transactionID to be of type string

### PR DESCRIPTION
**Pull request by nickcorin**
We're no longer enforcing the `uuid.UUID` type on `transactionID` as some products would prefer to generate a key in their own way. A `uuid` will still be valid by using `uuidVal.String()`.

### CODE REVIEW

File: pkg/types/chain_writer.go [hunk 1]
```diff
@@ -10,9 +10,9 @@ import (
 type ChainWriter interface {
 	// SubmitTransaction packs and broadcasts a transaction to the underlying chain.
 	//
-	// - `args` specifies input parameters to the contract call.
+	// - `args` should be any object which maps a set of method param into the contract and method specific method params.
 	// - `transactionID` will be used by the underlying TXM as an idempotency key, and unique reference to track transaction attempts.
```
Comment (ilija42) [inline pkg/types/chain_writer.go:14]: can you add to the description of the PR why UUID doesn't work?

Comment (nickcorin) [reply pkg/types/chain_writer.go:14]: UUID does work, and it's fine to send a UUID. It's just that it doesn't have to be enforced.

File: pkg/types/chain_writer.go [hunk 2]
```diff
@@ -10,9 +10,9 @@ import (
 type ChainWriter interface {
 	// SubmitTransaction packs and broadcasts a transaction to the underlying chain.
 	//
-	// - `args` specifies input parameters to the contract call.
+	// - `args` should be any object which maps a set of method param into the contract and method specific method params.
```
Comment (jmank88) [inline pkg/types/chain_writer.go:13]: Did we lose some work here? ChainReader has a much more comprehensive comment, and ChainWriter should either have the same, or we should be using a shared `type SomeAlias any` with the doc in one place.

**Review by jmank88 — APPROVED** [review]

File: pkg/types/chain_writer.go (hunk 2 above)
Comment (nickcorin) [reply pkg/types/chain_writer.go:13]: Yeah, the `args` param changed and the comment didn't get updated accordingly. Probably my fault.

Comment (nickcorin) [reply pkg/types/chain_writer.go:13]: I'll update the docs in a follow up PR.

**Review by ilija42 — APPROVED** [review]

**Review by silaslenihan — APPROVED** [review]

**Pull request merged**